### ISSUE TITLE
Bug: [require-await] reports class function with async prefix

### ISSUE THREAD

**Issue by trsh** [typescript-eslint/typescript-eslint#5799]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.8.4&jsx=true&sourceType=module&code=MYGwhgzhAEBiD29oG8CwAoa1IE8B2w0ARmAE4AUAlChllqQKYAuArqXtAAqnwC2AlhAYA6RhHggAbg3IBGSgG5a0AL4ZluAtABmiKjUx0wAdzD8m0JgAtBwkhUXK16ZxmDw8ECwGswv6AC80HgMxnB6jui+vsK68FQKQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEI0QR1gEtpEBaAQwHdzCAXdMAbUkWmgHtpIBdcMAXxD8gA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
class Foo {
  async bar() {
    return Promise.resolve(1);
  }

  async foo() {
    await this.bar();
  }
}

const kaka = new Foo();
kaka.foo();
```

### ESLint Config

```javascript
{
  "rules": {
    "require-await": ["error"]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

It should not report functions not having await, it doesn't make sense, but instead the calls who don't use await.

### Actual Result

It reports functions not having await.

### Versions

One in playground example

**Comment by bradzacher** [comment]
You ticked that you have read the FAQ, but I don't think you did because the very first FAQ answered your question.

https://typescript-eslint.io/docs/linting/troubleshooting#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

You're using the wrong rule.

https://typescript-eslint.io/rules/require-await
